src/utils.ts

```diff
@@ -39,7 +39,7 @@ export function isDefined<T extends any>(param: T | undefined | null): param is
   return param !== null && param !== undefined;
 }
 
-export type Comparable = string | number | bigint | Date | Temporal.PlainDate | Temporal.ZonedDateTime;
+export type Comparable = string | number | bigint | Date | Temporal.PlainDate | Temporal.ZonedDateTime | undefined;
```

Comment:
We check for `undefined` on the 1st line of `compare`'s impl, so seems like `undefined` was already allowed as a `Comparable`